app/controllers/admin/users_controller.rb

```diff
@@ -0,0 +1,69 @@
+class Admin::UsersController < ApplicationController
+  before_action :if_not_admin
+  before_action :set_user, only: [:show, :edit, :destroy, :update]
+
+  def index
+    @users = User.all
+  end
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      redirect_to admin_users_path(@user.id)
+    else
+      render :new
+    end
+  end
+
+  def show
+    @user = User.find(params[:id])
+    # @tasks = @user.tasks
+    @tasks = @user.tasks.select(:task_name)
+  end
+
+  def edit
+  end
+
+  def update
+    if @user.update(user_params)
+      redirect_to admin_users_url, notice: 'User was successfully updated.'
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    # @user.destroy
+    # @user.tasks.destroy
+    # redirect_to admin_users_url, notice: 'User was successfully destroyed.'
+
+    if @user.destroy
+      redirect_to admin_users_url, notice: 'User was successfully destroyed.'
+    else
+      # binding.irb
+      redirect_to admin_users_path, alert: "#{@user.errors[:base][0]}"
+    end
+  end
+
+
+  private
+    def if_not_admin
```

Comment:
メソッド名に条件式の内容を記載するのはあまり良くないですね。

理由としては、メソッド内の条件式の内容が変わった場合に、メソッド名も変える必要が出てしまい、もし条件の内容とメソッド名が統一されていない場合は、どちらが正しい条件なのか分からなくなってしまうためです。

命名の例として、「管理者以外のユーザーのアクセスを制限する」「別の画面に遷移させる」などの処理を読み替えて「管理者であることを要求する」というメソッド名を付けても良いかもしれません。
```suggestion
    def require_admin
```